### PR TITLE
Send a webhook event

### PR DESCRIPTION
**Pull request by kuzmany**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR added new dispatcher to send a webhook campaign action what allow for devs work  with wbhook response.

#### Steps to test this PR:
1. Install plugin https://github.com/kuzmany/mautic-extendee-webhook-response-bundle 
2. Use decision Response from webhook after Send a webhook

### CODE REVIEW

File: app/bundles/WebhookBundle/WebhookEvents.php [hunk 1]
```diff
@@ -98,4 +98,14 @@ final class WebhookEvents
...
+
+    /**
+     * The mautic.webhook.campaign_on_send_webhook event is dispatched when send a webhook campaign action is fired.
+     *
+     * The event listener receives a
+     * Mautic\CampaignBundle\Event\CampaignTriggerEvent instance.
+     *
+     * @var string
+     */
+    const ON_SEND_WEBHOOK = 'mautic.webhook.campaign_on_send_webhook';
```
Comment (alanhartless) [inline app/bundles/WebhookBundle/WebhookEvents.php]: Let's rename this event to match it's action as it's not dispatched when we "send" the webhook but when we are handling the response. `const ON_WEBHOOK_RESPONSE = 'mautic.webhook.on_response';`

**Review by alanhartless — DISMISSED** [review]

**Comment by alanhartless** [comment]
@kuzmany can you also handle the conflict? Thanks!

**Comment by npracht** [comment]
Hi there!
It has been decided to not create any extra Mautic 2 minor versions (which means no more features in M2) knowing that Mautic 3 is coming very soon.

**We now want to integrate your contribution in the Mautic 3 roadmap as 3.1.0 candidate.**

## How to do?
1. Check if your feature or enhancement is still missing and relevant in Mautic 3
2. **If Yes**: rebase your PR against the 3.x branch
3. **If No**: please consider closing your PR if you are absolutely sure that the feature has made it into Mautic 3 already

**Please report results by commenting on your PR to make us administration easier.**
______
You can more information on how to do all of that on [this blog post "Getting you PR ready for Mautic 3"](https://www.mautic.org/blog/developer/getting-your-pr-ready-mautic-3).

**Comment by ImCowboySibs** [comment]
Did this end up getting added to Mautic 2.16, or should we still use @kuzmany plugin or PR solution?

Thanks

**Comment by kuzmany** [comment]
Rebased

**Comment by RCheesley** [comment]
@kuzmany looks like we need to improve the test coverage on some of the files in this PR before we can merge. See https://github.com/mautic/mautic/pull/6873/files for some hints on the lines not covered.  Could you take a look at this so we can look at merging in 3.1?

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by dennisameling** [comment]
Hi @kuzmany! Thanks for your PR 🚀 we just upgraded Mautic to Symfony 4 and this PR will need to be updated. Since this PR has conflicts now, could you please do a `git merge upstream/features` locally to resolve the conflicts? Thanks in advance!

**Comment by julienWebmecanik** [comment]
I think there is dead code on this old branch regarding `WebHookBundle\WebHookEvents::ON_WEBHOOK_RESPONSE` and `WebHookBundle\CampaignSubscriber::onCampaignTriggerAction()`.
Can someone check ?

**Comment by kuzmany** [comment]
@julienWebmecanik it's not deadcode

https://github.com/kuzmany/mautic/blob/send-a-webhook-event/app/bundles/WebhookBundle/EventListener/CampaignSubscriber.php#L84

WebhookEvents::ON_CAMPAIGN_TRIGGER_ACTION - is triggered on fired campaign action to process webhook (example GET URL)
WebHookBundle\WebHookEvents::ON_WEBHOOK_RESPONSE is triggered during the campaign action when you process response from webhook (response from GET URL)

For example you can send data somewhere by this action and also process response from this webhook.

**Comment by kuzmany** [comment]
We do not need it anymore. If anybody need it, can cherry pick it

**Pull request closed without merging**